wai-aria/role/invalid-roles.html

```diff
@@ -11,26 +11,109 @@
-- escaped chars, URL-encoded chars
-- backslash closing quote and other obvious hack attempts
-- etc.
+<!-- Tests fallback for semantically neutral elements when 1/2/3 invalid roles are supplied -->
+<span role="foo" data-testname="span with invalid role name foo" class="ex-generic">x</span>
+<span role="foo bar" data-testname="span with two invalid role names" class="ex-generic">x</span>
+<span role="foo bar baz" data-testname="span with three invalid role names" class="ex-generic">x</span>
+<div role="foo" data-testname="div with invalid role name foo" class="ex-generic">x</div>
+<div role="foo bar" data-testname="div with two invalid role names" class="ex-generic">x</div>
+<div role="foo bar baz" data-testname="div with three invalid role names"class="ex-generic">x</div>
 
--->
+<!-- Whitespace tests with <nav> (including line breaks, tabs, zero-width space, braille space, non-breaking space, standard space) -->
+<nav role=" " data-testname="nav with empty character as role" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#13" data-testname="nav with line break (&#13) character as role" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#9" data-testname="nav with tab (&#9) as role (should be treated as whitespace)" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#8203" data-testname="nav with zero-width space as role (should be treated as whitespace)" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#10240" data-testname="nav with braille space (10240) as role" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#x2800" data-testname="nav with braille space (x2800) as role" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&nbsp" data-testname="nav with non-breaking space (nbsp) as role" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#20" data-testname="nav with standard space (nbsp) as role" data-expectedrole="navigation" class="ex">x</nav>
+
+<!-- Whitespace tests with <span> (including line breaks, tabs, zero-width space, braille space, non-breaking space, standard space) -->
+<span role=" " data-testname="span with empty character as role" class="ex-generic">x</span>
+<span role="&#13" data-testname="span with line break (&#13) character as role" class="ex-generic">x</span>
+<span role="&#9" data-testname="span with tab (&#9) as role (should be treated as whitespace)" class="ex-generic">x</span>
+<span role="&#8203" data-testname="span with zero-width space as role (should be treated as whitespace)" class="ex-generic">x</span>
+<span role="&#10240" data-testname="span with braille space (10240) as role" class="ex-generic">x</span>
+<span role="&#x2800" data-testname="span with braille space (x2800) as role" class="ex-generic">x</span>
+<span role="&nbsp" data-testname="span with non-breaking space (nbsp) as role" class="ex-generic">x</span>
+<span role="&#20" data-testname="span with standard space (nbsp) as role" class="ex-generic">x</span>
```

Comment:
If we're trying to break the parser, don't be so polite as to escape everything. 😉

Keep what you have, but duplicate the tests with unescaped variants, too.